Add initial TF-M support for nRF54L10

Vge0rge opened the pull request:
Please read commit messages for details.
This brings limited support for TF-M with the nRF54L10 in Zephyr and it was tested using two tfm_intergation samples.

Vge0rge committed:
modules: TF-M: nordic: Update header for the NRF54L series

Update the header to include a generic header for the NRF54L
series and not a specific one for the NRF54L15.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
boards: nordic: Add initial support for nRF54L10 with TF-M

This adds the nrf54l15dk/nrf54l10/cpuapp/ns board variant to
Zephyr. It allows to build applications for this target.

This is an initial support for the non secure target which allows
building and running tfm_ipc and config_build.

This is NOT full support of the non secure target in upstream
Zephyr.

There are important limitations, such as:
  - The hardware Crypto accelerator is not supported and thus the non
    secur target is NOT secure for production applications in upstream
    Zephyr.
  - The BL2 is not supported, so no DFU is supported with this support

Most of the code changes here are taken from nRF Connect SDK
in order to avoid having noups there.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
samples: tfm: Add nRF54L10 in TF-M samples yaml

Add nRF54L10 non secure target for the config_build and
the tfm_ipc samples.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
samples: tests: Exclude nRF54L10 NS for unsupported tests/samples

The ADC and watchdog samples are not currently supported for the
non secure target of the nRF54L10 DK.

Add exclusions for these until they are supported.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
manifest: Bring TF-M with initial nRF54L10 support

Brings some commits from the upstream TF-M which introduce
nRF54L10 ns support.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

d3zd3z reviewed (dismissed):
The change itself looks good, although I suspect we'll end up failing due to an issue with MbedTLS 3.6.3.

tomi-font commented on boards/nordic/nrf54l15dk/Kconfig:
```diff
@@ -3,7 +3,7 @@
 # nRF54L15 DK board configuration
 
-if BOARD_NRF54L15DK_NRF54L15_CPUAPP_NS
+if BOARD_NRF54L15DK_NRF54L15_CPUAPP_NS || BOARD_NRF54L15DK_NRF54L10_CPUAPP_NS
```
Typos in this commit:
- secur**e** [...] applicati**~~i~~**ons
- cha**gn**es

tomi-font commented on boards/nordic/nrf54l15dk/board.cmake:
```diff
@@ -10,7 +10,7 @@ elseif(CONFIG_SOC_NRF54L05_CPUFLPR OR CONFIG_SOC_NRF54L10_CPUFLPR)
 endif()
 
-if(CONFIG_BOARD_NRF54L15DK_NRF54L15_CPUAPP_NS)
+if(CONFIG_BOARD_NRF54L15DK_NRF54L15_CPUAPP_NS OR CONFIG_BOARD_NRF54L15DK_NRF54L10_CPUAPP_NS)
```
How about using `CONFIG_BUILD_WITH_TFM` instead?

Vge0rge replied:
I think that this should be possible but all the other Nordic boards seem to use this methodology. I would say to keep it as it and in a separate PR we can change the behavior for all of them, not for on board.

tomi-font commented on boards/nordic/nrf54l15dk/nrf54l15dk_nrf54l10_cpuapp_ns.dts:
```diff
@@ -0,0 +1,113 @@
+	/* Disable so that TF-M can use this UART */
+	status = "disabled";
+
+	current-speed = <115200>;
+	pinctrl-0 = <&uart30_default>;
+	pinctrl-1 = <&uart30_sleep>;
+	pinctrl-names = "default", "sleep";
```
Are those lines needed if the UART is disabled?

Vge0rge replied:
You are right, these can be removed.

tomi-font commented on boards/nordic/nrf54l15dk/nrf54l15dk_nrf54l10_cpuapp_ns.yaml:
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
`Apache-2.0`?

tomi-font commented on boards/nordic/nrf54l15dk/nrf54l15dk_nrf54l10_cpuapp_ns_defconfig:
```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
`Apache-2.0`?

tomi-font commented on modules/trusted-firmware-m/nordic/nrf54l10_cpuapp/CMakeLists.txt:
```diff
@@ -0,0 +1,23 @@
+
+set(NRF_BOARD_SELECTED True)
+
+add_subdirectory(${Trusted\ Firmware\ M_SOURCE_DIR}/platform/ext/target/nordic_nrf/common/nrf54l10 nrf54l10)
```
Is `nrf54l10` duplicated here or is it normal?

Vge0rge replied:
Thats normal, the second part is the binary directory (as they call it in cmake). In short it is the name of the directory inside the build folder and it cannot be removed because of how TF-M support is at the moment.

tomi-font reviewed (changes requested):
~~`BOARD_NRF54L15DK_NRF54L10_CPUAPP_NS` is used but it's not defined anywhere?~~ Nevermind, seems to be how it's done.

Vge0rge committed:
boards: nordic: Update copyrights for nRF54L15

It seems that some copyrights were not udpated to
Apache-2.0 as they should have been so this commit
updates them.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

nordicjm commented on boards/nordic/nrf54l15dk/nrf54l15dk_nrf54l10_cpuapp_ns.dts:
```diff
@@ -0,0 +1,108 @@
+			reg = <0x20012000 DT_SIZE_K(72)>;
+		};
+	};
+
```
```suggestion
```

nordicjm reviewed (changes requested):
minor nit.
